### ISSUE TITLE
Configured DNS but iotedge check reports a DNS warning

### ISSUE THREAD

**Issue by fgheysels** [Azure/iotedge-eflow#246]
**Describe the bug**

I have deployed EFLOW on Windows 2019 Core.
I have configured DNS for Eflow by using these commands:

```
$dnsServerIPs = Get-DnsClientServerAddress -AddressFamily IPv4 -InterfaceAlias "Ethernet" 
$eflowEndpoint = Get-EflowVmEndpoint
Set-EflowVmDNSServers -vEndpointName $eflowEndpoint.Name -dnsServers $dnsServerIPs.ServerAddresses
```

Afterwards, I connected to the EFLOW vm and executed the `iotedge check` command.  To my surprise, a warning is issued saying DNS is not configured:

![image](https://user-images.githubusercontent.com/3605786/235130114-2b55638e-0116-429c-9bf7-55151b7f6148.png)

When visiting the reported [site](https://learn.microsoft.com/en-us/azure/iot-edge/troubleshoot-common-errors?view=iotedge-1.4#solution-2), I checked the  `/etc/docker/daemon.json` file, and indeed, there are no DNS entries there.

However, I also looked at [this site](https://learn.microsoft.com/en-us/azure/iot-edge/troubleshoot-iot-edge-for-linux-on-windows-networking?view=iotedge-1.4#check-domain-name-system-dns-configuration) where DNS for eflow is more explained.

I've checked the `/etc/systemd/resolved.conf` file and cannot find anything related to the configured DNS server:

![image](https://user-images.githubusercontent.com/3605786/235130792-611cb4a5-c3fb-4967-867d-f3ec007457a3.png)

So, it looks like DNS is indeed not configured ?

Does eflow take into account the daemon.json file ?
Should the `Set-EflowVmDNSServers` write DNS info to the daemon.json file ?

**Comment by jagadishmurugan** [comment]
@fgheysels ,
Can you try the "resolvectl" cmd and share the output. You should see the DNS server(s) configured as part of Set-EflowVMDNSServer cmd in the output...

Regards,
Jagadish

**Comment by jagadishmurugan** [comment]
@fgheysels ,
We dont add DNS to docker daemon.json file. But you can add it as appropriate to your scenario.

**Comment by jagadishmurugan** [comment]
@veyalla ,
can you take a look?

**Comment by veyalla** [comment]
>It would be good if iotedge check would not report that DNS is not configured in this case though.

Please add a https://aka.ms/iotedge-suggestion

**Comment by jagadishmurugan** [comment]
@fgheysels , this is not really EFLOW related. can you close this issue?

**Comment by fgheysels** [comment]
Yes I can, but it is maybe an idea that the `Set-EflowVmDnsServers` cmdlet also modifies the daemon.json file ?

**Comment by fgheysels** [comment]
Created an IoT Edge suggestion [here](https://feedback.azure.com/d365community/idea/effa2592-7fea-ed11-a81c-000d3a7a9cdb)
